core/src/main/java/org/apache/accumulo/core/data/LoadPlan.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.data;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.apache.accumulo.core.client.admin.TableOperations.ImportDestinationOptions;
+import org.apache.hadoop.io.Text;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.UnsignedBytes;
+
+/**
+ * Information about where to load files into an Accumulo table.
+ *
+ * @see ImportDestinationOptions#usingPlan(LoadPlan)
+ * @since 2.0.0
+ */
+public class LoadPlan {
+  private final ImmutableList<Destination> destinations;
+
+  private static byte[] copy(byte[] data) {
+    return data == null ? null : Arrays.copyOf(data, data.length);
+  }
+
+  private static byte[] copy(Text data) {
+    return data == null ? null : data.copyBytes();
+  }
+
+  private static byte[] copy(CharSequence data) {
+    return data == null ? null : data.toString().getBytes(StandardCharsets.UTF_8);
+  }
+
+  private static String checkFileName(String fileName) {
+    Preconditions.checkArgument(Paths.get(fileName).getNameCount() == 1,
+        "Expected only filename, but got %s", fileName);
+    return fileName;
+  }
+
+  /**
+   * @since 2.0.0
+   */
+  public enum RangeType {
+    /**
+     * Range that correspond to one or more tablets in a table. For a range of this type the start
+     * row and end row can be null. The start row is exclusive and the end row is inclusive (like
+     * Accumulo tablets). A common use case for this would be when files were partitioned using a
+     * tables splits. When using this range type the start and end row must exist as splits in the
+     * table or an exception will be thrown at load time.
+     */
+    TABLET,
```

Comment:
Could make this name something like "splits" since its likely that is where they will fall.  I know names are difficult!